plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -695,6 +704,34 @@ public void renameSchema(ConnectorSession session, String schemaName, String new
         return properties.buildOrThrow();
     }
 
+    private static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, index) -> {
+            // Convert to LocalDate from java.sql.Date via String because java.sql.Date#toLocalDate() returns wrong results in B.C. dates. -5881579-07-11 -> +5881580-07-11
+            // Phoenix JDBC driver supports getObject(index, LocalDate.class), but it leads to incorrect issues. -5877641-06-23 -> 7642-06-23 & 5881580-07-11 -> 1580-07-11
+            // Still returns +10 days during julian -> gregorian switch
```

Comment:
Unclear. Do you mean that current implementation still returns +10 days during switch or does it refer to `getObject(index, LocalDate.class)`?

Reply:
The former. Updated the code comment.